_pytest/python.py

```diff
@@ -889,13 +892,22 @@ def parametrize(self, argnames, argvalues, indirect=False, ids=None,
         if scope is None:
             scope = "function"
         scopenum = scopes.index(scope)
+        valtypes = dict.fromkeys(argnames, "funcargs")
```

Comment:
This section needs to be tidied up a bit. I think it would be simpler if you first deal with setting the right values in the valtypes dict, based on the value of indirect.
Then, secondly, go through that dict and check if we need to raise the "uses no fixture" exception.
Also, rather than checking if indirect is a tuple or list, it might be nicer to explicitly check for True and False, else assume it is a sequence and iterate over it.